packages/hooks/src/hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react";
+
+/***
+ * Return true if given media query string evaluates to true.
+ *
+ * Note: the initial server-side rendering will always return false.
+ * After that the variable is updated by event listeners on server-side.
+ */
+export const useMediaQuery = (query: string): boolean => {
+  const [matchesQuery, setMatchesQuery] = useState(false);
+  const checkMatch = () => setMatchesQuery(window.matchMedia(query).matches);
+
+  useEffect(() => {
+    // Set initial state, which will always be false for the initial,
+    // server-side rendering.
+    checkMatch();
+
+    // Update the state whenever the component is (re)rendered in browser.
+    window.addEventListener("resize", checkMatch);
+    return () => window.removeEventListener("resize", checkMatch);
+  }, []);
```

Comment:
This should include a dependency for `checkMatch`

Reply:
The idea was to run the effect only once, hence no dependencies. Although, `checkMatch` being defined as it is, I'm not sure it will trigger reruns anyway. Based on a quick test, adding `checkMatch` to dependencies works as well, but it was hard to say whether this caused extra renders or not. Thoughts?

Reply:
It should functionally be the same, but since `checkMatch` is used in the effect, proper practice is to include it in the dependencies too (some linters even notify about this). I don't think it matters a lot in practice here, but  it is more future proof in case the hook evolves into something more complex.

Reply:
[Fixed](https://github.com/thunderstore-io/thunderstore-ui/pull/332/commits/d7b58111663d7dd9efc13fb316bd521733f6aef7#diff-c603c0226e1474d16fae89b64097f6f7a19fff80b1c0dd728d6ccdf1c7cc2d6dR21)